src/Integration/NewConnectedMode/ConfigurationSerializer.cs

```diff
@@ -0,0 +1,108 @@
+/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Alm.Authentication;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.Integration.Persistence;
+
+namespace SonarLint.VisualStudio.Integration.NewConnectedMode
+{
+    /// <summary>
+    /// Reads/write the binding configuration for the new connected mode.
+    /// The legacy connected mode binding is written using <see cref="SolutionBindingSerializer"/>.
+    /// </summary>
+    internal class ConfigurationSerializer : ISolutionBindingSerializer
```

Comment:
I decided to reuse the existing _ISolutionBindingSerializer_ to read/write the new config.
Currently only the _read_ part is implemented, and it looks for a file called _.sonarlint\\[SOLUTIONFILENAME].sqconfig_